Make isolated graph builds fail deterministically on undeclared cross-project targets (#14280)
dotnet/msbuild: documentation/wiki/ChangeWaves.md

- [XmlPeek, XmlPoke, and XslTransformation default to prohibiting embedded DTDs](https://github.com/dotnet/msbuild/pull/14285)
- [Out-of-proc communication uses larger read-ahead buffers and pre-buffers TaskHost packet bodies to reduce pipe reads.](https://github.com/dotnet/msbuild/pull/14505)
- [Restore no longer discards a ProjectRootElementCache that reloads changed files from disk, so the build that follows an implicit restore does not re-parse the import closure.](https://github.com/dotnet/msbuild/pull/14558)

> - [Isolated (`-graph -isolate`) builds fail deterministically with MSB4252 on a cross-project reference to a target that is not declared via `ProjectReferenceTargets`, instead of passing or failing depending on MSBuild node scheduling. Set `MSBUILDDISABLEFEATURESFROMVERSION=18.11` to restore the prior (scheduling-dependent) behavior.](https://github.com/dotnet/msbuild/pull/14280)

Reviewer: We need to figure out how to convince the robot to stop putting this redundant instruction in everything . . .
```suggestion
- [Isolated (`-graph -isolate`) builds fail deterministically with MSB4252 on a cross-project reference to a target that is not declared via `ProjectReferenceTargets`, instead of passing if the referenced project happened to build on that node previously.](https://github.com/dotnet/msbuild/pull/14280)
```

Reviewer 2: That is true!
